aten/src/ATen/core/ivalue_inl.h

```diff
@@ -524,6 +534,26 @@ struct ivalue::PyObjectHolder : c10::intrusive_ptr_target {
   virtual ~PyObjectHolder() {};
 };
 
+struct ivalue::EnumHolder : c10::intrusive_ptr_target {
+ public:
+  EnumHolder(std::string qualified_class_name, std::string name, IValue value)
+      : qualified_class_name_(std::move(qualified_class_name)),
+        name_(std::move(name)), value_(std::move(value)) {}
+
+  bool is(const ivalue::EnumHolder& rhs) {
+    return *this == rhs;
+  }
+
+  bool operator==(const ivalue::EnumHolder& o) {
+    return qualified_class_name_ == o.qualified_class_name_ &&
+        name_ == o.name_ && value_ == o.value_;
+  }
+
+  std::string qualified_class_name_;
+  std::string name_;
```

Comment:
consider storing the name_ as an interned symbol.

Reply:
Yeah I guess it makes sense but also feels like a bit of overkill. Enum names usually aren't frequently used, so interning it doesn't seem to save much. What do you think?

Reply:
Yea, i think it's not worth it